src/WorkflowManager/Storage/Services/DicomService.cs

```diff
@@ -90,12 +89,43 @@ public async Task<PatientDetails> GetPayloadPatientDetailsAsync(string payloadId
             return patientDetails;
         }
 
-        public async Task<string?> GetFirstValueAsync(IList<VirtualFileInfo> items, string payloadId, string bucketId, string keyId)
+        private string? GetFirstValueAsync(Dictionary<string, DicomValue>? dict, string keyId)
         {
-            ArgumentNullException.ThrowIfNullOrWhiteSpace(bucketId, nameof(bucketId));
-            ArgumentNullException.ThrowIfNullOrWhiteSpace(payloadId, nameof(payloadId));
             ArgumentNullException.ThrowIfNullOrWhiteSpace(keyId, nameof(keyId));
+            if (dict is null)
+            {
+                return null;
+            }
+
+            try
+            {
+                if (dict is null)
```

Comment:
Any reason why we're checking if dict is null twice?

Reply:
Correct :) removed